app/src/main/java/de/nico/ha_manager/activities/Main.java

```diff
@@ -67,6 +68,7 @@ public boolean onOptionsItemSelected(MenuItem item) {
         switch (item.getItemId()) {
             case R.id.action_settings:
                 startActivity(new Intent(this, Preferences.class));
+				finish();

```

Comment:
This finishes the activity, and with some `onBackPressed()` magic it will relaunch the activity from the Preferences.